typescript/sdk/src/router/adapters/EvmRouterAdapter.ts

```diff
@@ -7,76 +7,68 @@ import {
-    ContractAddrs extends RouterAddress = RouterAddress,
-  >
-  extends BaseEvmAdapter<ContractAddrs>
-  implements IRouterAdapter<ContractAddrs>
-{
-  interchainSecurityModule(chain: ChainName): Promise<Address> {
-    return this.getConnectedContract(chain).interchainSecurityModule();
+export class EvmRouterAdapter extends BaseEvmAdapter implements IRouterAdapter {
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider<any>,
+    public readonly addresses: { router: Address },
+  ) {
+    super(chainName, multiProvider);
+  }
+
+  interchainSecurityModule(): Promise<Address> {
+    return this.getConnectedContract().interchainSecurityModule();
   }
 
-  owner(chain: ChainName): Promise<Address> {
-    return this.getConnectedContract(chain).owner();
+  owner(): Promise<Address> {
+    return this.getConnectedContract().owner();
   }
 
-  remoteDomains(originChain: ChainName): Promise<Domain[]> {
-    return this.getConnectedContract(originChain).domains();
+  remoteDomains(): Promise<Domain[]> {
+    return this.getConnectedContract().domains();
   }
 
-  async remoteRouter(
-    originChain: ChainName,
-    remoteDomain: Domain,
-  ): Promise<Address> {
-    const routerAddressesAsBytes32 = await this.getConnectedContract(
-      originChain,
-    ).routers(remoteDomain);
+  async remoteRouter(remoteDomain: Domain): Promise<Address> {
+    const routerAddressesAsBytes32 = await this.getConnectedContract().routers(
+      remoteDomain,
+    );
     return bytes32ToAddress(routerAddressesAsBytes32);
   }
 
-  async remoteRouters(
-    originChain: ChainName,
-  ): Promise<Array<{ domain: Domain; address: Address }>> {
-    const domains = await this.remoteDomains(originChain);
+  async remoteRouters(): Promise<Array<{ domain: Domain; address: Address }>> {
+    const domains = await this.remoteDomains();
     const routers: Address[] = await Promise.all(
-      domains.map((d) => this.remoteRouter(originChain, d)),
+      domains.map((d) => this.remoteRouter(d)),
     );
     return domains.map((d, i) => ({ domain: d, address: routers[i] }));
   }
 
-  getConnectedContract(chain: ChainName): Router {
-    const address = this.multiProvider.getChainMetadata(chain).router;
-    const provider = this.multiProvider.getEthersV5Provider(chain);
-    return Router__factory.connect(address, provider);
+  getConnectedContract(): Router {
+    return Router__factory.connect(this.addresses.router, this.getProvider());
   }
 }
 
-export class EvmGasRouterAdapter<
-    ContractAddrs extends RouterAddress = RouterAddress,
-  >
-  extends EvmRouterAdapter<ContractAddrs>
-  implements IGasRouterAdapter<ContractAddrs>
+export class EvmGasRouterAdapter
+  extends EvmRouterAdapter
+  implements IGasRouterAdapter
 {
-  async quoteGasPayment(
-    origin: ChainName,
-    destination: ChainName,
-  ): Promise<string> {
+  async quoteGasPayment(destination: ChainName): Promise<string> {
```

Comment:
string?

Reply:
Different providers (ethers5, 6, viem, sol web3) use different BigNumber types. String is the common denominator.

We could alternatively normalize everything to `BigInt` ourselves if you prefer, but we may need to polyfill for old browsers/node versions. This was just easiest